### PR TITLE
[SQL] Support SHOW CHARACTER SET Statement

### LINKED ISSUE #388: [SQL] Support `SHOW CHARACTER SET Statement`
**What would you like to be added**:

**Why is this needed**:

https://dev.mysql.com/doc/refman/8.0/en/show-character-set.html

### PR DESCRIPTION
**Pull request by robbinhan**
**What this PR does**:
Support SHOW CHARACTER SET Statement

**Which issue(s) this PR fixes**:
Fixes #388 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/runtime/optimize/dal/show_character_set.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * limitations under the License.
+ */
+
+package dal
+
+import (
+	"context"
+
+	"github.com/arana-db/arana/pkg/proto"
+	"github.com/arana-db/arana/pkg/proto/rule"
```
Comment (dongzl) [inline pkg/runtime/optimize/dal/show_character_set.go:24]: Please format the import. use this tool:
```
go get -u github.com/dubbogo/tools/cmd/imports-formatter

imports-formatter
```

File: pkg/runtime/plan/dal/show_character_set.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * limitations under the License.
+ */
+
+package dal
+
+import (
+	"context"
+	"strings"
+
+	"github.com/arana-db/arana/pkg/proto"
```
Comment (dongzl) [inline pkg/runtime/plan/dal/show_character_set.go:24]: The same.

**Review by dongzl — CHANGES_REQUESTED** [review]

File: pkg/runtime/ast/show.go [hunk 3]
```diff
@@ -463,3 +463,25 @@ func (s *ShowWarnings) Restore(flag RestoreFlag, sb *strings.Builder, args *[]in
...
+	*baseShow
+}
+
+func (s *ShowCharset) Mode() SQLType {
+	return SQLTypeShowCharacterSet
+}
+
+func (s *ShowCharset) Validate() error {
+	return nil
+}
```
Comment (jjeffcaii) [inline pkg/runtime/ast/show.go:477]: Delete this method, no need any more.

**Review by jjeffcaii — COMMENTED** [review]
Pls add UT.

**Review by dongzl — APPROVED** [review]

**Review by jjeffcaii — APPROVED** [review]

**Pull request merged**